plugins/eventMacro/eventMacro/Runner.pm

```diff
@@ -216,48 +222,103 @@ sub overrideAI {
-		debug "[eventMacro] No need to add/clear AI_queue because it already is compatible with this macro.\n", "eventMacro", 2;
-		return;
+		$eventMacro->{Macro_Runner}{$self->{slot}}->last_subcall_overrideAI($overrideAI);
 	}
 	
-	if ($overrideAI == 0) {
-		$self->register;
+	# Current macro should not be in AI queue
+	if ($overrideAI == 1) {
 		
-	} elsif ($overrideAI == 1) {
-		$self->unregister;
+		# There is an eventMacro sequence in AI queue
+		if (defined $macro_index) {
+			my $macro_args = AI::args($macro_index);
+			
+			# And it contains the current macro
+			if (exists $macro_args->{$self->{slot}}) {
+			
+				# If this is the only macro in the eventMacro sequence remove it from the AI queue
+				if (keys %{$macro_args} == 1) {
+					$self->unregister;
+				
+				# If there are more macros in the eventMacro sequence only remove this one
+				} else {
+					$self->register_remove($macro_args);
+				}
+			} else {
+				debug "[eventMacro] No need to add/clear AI_queue because it already is compatible with this macro.\n", "eventMacro", 2;
+				return;
+			}
+			
+		} else {
+			debug "[eventMacro] No need to add/clear AI_queue because it already is compatible with this macro.\n", "eventMacro", 2;
+			return;
+		}
+	
+	} else {
+	
+		if (defined $macro_index) {
+			my $macro_args = AI::args($macro_index);
+			
+			if (exists $macro_args->{$self->{slot}}) {
+				debug "[eventMacro] No need to add/clear AI_queue because it already is compatible with this macro.\n", "eventMacro", 2;
+				return;
+			} else {
+				$self->register_add($macro_args);
+			}
+			
+		} else {
+			$self->register_new;
+		}
 	}
 }
 
-# Registers to AI queue
-sub register {
+# Registers new eventMacro sequence to AI queue
+sub register_new {
 	my ($self) = @_;
-	debug "[eventMacro] Macro '".$self->{name}."' is now registering itself to AI queue.\n", "eventMacro", 2;
+	debug "[eventMacro] Macro '".$self->{name}."' is now adding eventMacro to the AI queue sequence.\n", "eventMacro", 2;
 	if (AI::is("NPC")) {
 		splice(@AI::ai_seq, 1, 0, 'eventMacro');
-		splice(@AI::ai_seq_args, 1, 0, {});
+		splice(@AI::ai_seq_args, 1, 0, { $self->{slot} => $self->{name} });
 	} else {
-		AI::queue('eventMacro');
+		AI::queue('eventMacro', { $self->{slot} => $self->{name} });
 	}
 	$self->{registered} = 1;
 }
 
-# Unregisters from AI queue
+# Unregisters eventMacro sequence from AI queue
 sub unregister {
 	my ($self) = @_;
-	debug "[eventMacro] Macro '".$self->{name}."' is now deleting itself from AI queue.\n", "eventMacro", 2;
+	debug "[eventMacro] Macro '".$self->{name}."' is now deleting eventMacro from AI queue.\n", "eventMacro", 2;
 	AI::clear('eventMacro');
 	$self->{registered} = 0;
 }
 
+# Adds the current macro to the already existent eventMacro sequence in AI queue
+sub register_add {
+	my ($self, $macro_args) = @_;
+	debug "[eventMacro] Macro '".$self->{name}."' is now adding itself to the eventMacro sequente in AI queue.\n", "eventMacro", 2;
```

Comment:
`sequente` => `sequence`

Reply:
Make this message match the phrasing in `register_new`? Same for `unregister` and `register_remove`.